### PR TITLE
Make odbc namespace nested (odbc-cpp-wrapper)

### PR DESCRIPTION
**Pull request by mrylov**
## Description

This is a follow-up change of a review done for the HANA GDAL driver, see https://github.com/OSGeo/gdal/pull/5107#issuecomment-1025531581.
It has been suggested to rename the odbc namespace of the odbc-cpp library to qgs::odbc. This allows avoiding possible symbols conflicts of two instances of the odbc-cpp library used in QGIS and GDAL.

### CODE REVIEW

**Comment by rouault** [comment]
The proposed solution to modify the QGIS copy doesn't seem good to be able to maintain the qgs:: and generic (odbc::) version in the long term. The QGIS copy should just be a copy of the upstream project.
I would rather modify the upstream project to (ab)use #define magic to be able to configure the namespace, that would default to odbc:: , and could be overriden in one QGIS specific place to qgs::odbc.
See https://github.com/OSGeo/PROJ/blob/master/include/proj/util.hpp#L94 for a potential implementation (although I know some people don't like it because of the use of #define magic, but I'm not sure how to have customizable namespaces in C++ in a clean way)

**Comment by mrylov** [comment]
@rouault, thanks. Your proposal is fine to us and I will make necessary changes in https://github.com/SAP/odbc-cpp-wrapper.
Then, I will upload a new patch set for this pull request.

**Comment by mrylov** [comment]
@rouault, I've changed the code as you suggested. Please review the pull request again.

File: src/providers/hana/CMakeLists.txt [hunk 1]
```diff
@@ -81,8 +81,20 @@ add_library(provider_hana_a STATIC ${HANA_SRCS} ${HANA_HDRS})
...
+        "NS_ODBC_START=namespace qgs { namespace odbc {"
+        "NS_ODBC_END=} }"
+)
+target_compile_definitions(provider_hana_a
+    PUBLIC
+        ODBC_STATIC
+        "NS_ODBC=qgs::odbc"
+        "NS_ODBC_START=namespace qgs { namespace odbc {"
+        "NS_ODBC_END=} }"
+)
```
Comment (rouault) [inline src/providers/hana/CMakeLists.txt:97]: I guess the following should automatically retrieve the definitions from the odbccpp_static target, and also switch them to PRIVATE as they shouldn't be needed by users of providers_hana / providers_hana_a

```suggestion
target_compile_definitions(provider_hana PRIVATE $<TARGET_PROPERTY:odbccpp_static,INTERFACE_COMPILE_DEFINITIONS>)
target_compile_definitions(provider_hana_a PRIVATE $<TARGET_PROPERTY:odbccpp_static,INTERFACE_COMPILE_DEFINITIONS>)
```

**Review by rouault — COMMENTED** [review]
LGTM with the suggestion I made

File: src/providers/hana/CMakeLists.txt (hunk 1 above)
Comment (mrylov) [reply src/providers/hana/CMakeLists.txt]: Thanks. Done in 4aa326c12bc3a549d9edd42edfb8eeb447ebbc7c.

**Comment by mrylov** [comment]
@nyalldawson, could you please take a look at this pull request? Thanks.

**Comment by mrylov** [comment]
@nyalldawson, @lbartoletti, @3nids, could anyone of you review/approve this pull request?

**Review by rouault — APPROVED** [review]

**Pull request merged**